example_xla.py

```diff
@@ -159,11 +163,12 @@ def mp_main(
     dim: int = 4096,
     n_layers: int = 32,
     n_heads: int = 32,
+    prompt_len: int = 6,
```

Comment:
I set the default to 6 because I wanted the default to be the same number of input tokens as our previous prompt "I believe the meaning of life is", only I miscounted (that has 7 words), and there's not a 1-1 mapping between words and tokens necessarily, so it's still wrong regardless. We can change the default if the current one is not right - any suggestions on alternatives?

If I'm reading the code right, `max_batch_size` is related to the total number of prompts, not the number of tokens in each prompt. The goal of the `prompt_len` is to allow the user to specify a variable number of input tokens in a single prompt. (It is reasonable to point out that this does not currently support multiple prompts.) I'm not sure I understand your comments about `bs = 1` or about whether this is the right solution. (Whether what is the right solution?) Could you please clarify?

Reply:
Okay, now I get it. But from the discussion in gchat, do you still need this approach?

Reply:
If you mean our discussion about `max_seq_len` and `max_gen_len`, this is something quite separate, right? This PR allows the user to modify the length of the input prompt. `max_seq_len` controls the size allocated for the output (and in our repo, the total number of tokens generated), and `max_gen_len` controls the number of tokens displayed. I think we still need this for the user to modify input - please let me know if you had some other discussion in mind.